### ISSUE TITLE
Folders cointaining only `.gitignore`d files are sometimes not hidden, disappear only on click

### ISSUE THREAD

**Issue by bersbersbers** [microsoft/vscode#210341]
`coverage.py`'s `htmlcov` output folder contains a `.gitignore` file like this:

```
# Created by coverage.py
*
```

I am using `"explorer.excludeGitIgnore": true`, and so this folder should be hidden. Except, it isn't always, until I click it.

Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.88.1
- OS Version: Windows 11 23H2

Steps to Reproduce:

On Windows:
```batch
if exist bug rmdir /s/q bug
mkdir bug
for /l %i in (1,1,20) do (mkdir bug\sub%i && echo *> bug\sub%i\.gitignore)
code.cmd --disable-extensions bug
```

![Code_dOWoC85rl5](https://github.com/microsoft/vscode/assets/12128514/60e72db5-5560-488d-88e2-401deeedb67d)

This issue does not seem to reproduce if I replace `20` with `1` above.

**Comment by bersbersbers** [comment]
This issue has received the "new release" label, but I have been seeing it for months at least.

**Comment by lramos15** [comment]
Yeah it's the same cause as https://github.com/microsoft/vscode/issues/66971

Basically the folders are lazily resolved when you click on them, so once it's clicked and the contents are seen as both the root and all children are ignored the full folder is hidden
